### DISCUSSION TITLE
How do you merge multiple `Cypress.Chainable`?

### DISCUSSION THREAD

**Discussion by matthew-a-thomas** [cypress-io/cypress#21232] [Questions and Help]
I need to instantiate an [nspell](https://www.npmjs.com/package/nspell) object so that I can check the spelling of my app's pages. To do so I have to load three files from disk (an en-us .dic file, an en-us .aff file, and a personal .dic file). I know I can make individual chains for reading the individual files like this:
```javascript
let enUsAff = cy.readFile('en-us.aff');
let enUsDic = cy.readFile('en-us.dic');
let personalDictionary = cy.readFile('dictionary.dic');
```

But how do I make a chain that spits out an `nspell` instance once all three have completed?
```javascript
let nspell = cy.all( // If only this were a thing!
  enUsAff,
  enUsDic,
  personalDictionary
).then((aff, dic, personal) => NSpell(aff, dic).personal(personal));
```

**Comment by matthew-a-thomas** [comment]
Turns out that while it's not pretty, it's not difficult to just deeply nest the chains:
```javascript
cy.fixture('en_US.aff').then(aff =>
  cy.fixture('en_US.dic').then(dic =>
    cy.fixture('dictionary.dic').then(personal => {
      const nSpell = NSpell({ aff: aff, dic: dic }).personal(personal);
      cy.get('body').then(body => {
        const text = body.text();
        // assert correct spelling here...
      });
    })
  )
);
```

This can be encapsulated as a function for reuse.

**Comment by muratkeremozcan** [comment]
Here are 2 examples:

https://github.com/muratkeremozcan/fastify-example-tests/blob/main/cypress/integration/bonus03.js

https://github.com/muratkeremozcan/fastify-example-tests/blob/main/cypress/integration/spec08.js

Both from Gleb's course https://cypress.tips/courses

**Reply by matthew-a-thomas** [reply]
Could you point me to the exact lines you're talking about? I only see examples of deeply nesting the chains, although I was only skimming

**Reply by muratkeremozcan** [reply]
Just take a look at how the JSON files are being imported and used.

**Reply by matthew-a-thomas** [reply]
Ah. So are you saying that I can just replace e.g. `let enUsAff = cy.readFile('en-us.aff');` with `import * as enUsAff from 'en-us.aff'`, implying that I can avoid my trouble by just avoiding Cypress's chainable objects? If so there are two problems: my dictionary files are not JSON so they cannot be imported like that; and even if it did work then that would not be an example of merging multiple `Cypress.Chainable`, unless you're referring to how one chain is instantiated inside the `.then` block of another, in which case it's not materially different than the solution I already found and posted above.

**Reply by muratkeremozcan** [reply]
Right. 

I could give it a try if you could share a repo

**Reply by matthew-a-thomas** [reply]
Thank you for offering. I'm going to pass; I appreciate your help and would welcome more, but I've found a workaround that works in my case and there are bigger fish to fry. Thank you!
